platform-hub-api/db/migrate/20171012110416_import_kubernetes_tokens.rb

```diff
@@ -0,0 +1,94 @@
+class ImportKubernetesTokens < ActiveRecord::Migration[5.0]
+
+  BATCH_SIZE = 100
+
+  def up
+    # Start fresh before import
+    KubernetesToken.delete_all
+
+    # Cluster lookup
+    clusters = KubernetesCluster.all.reduce({}) {|acc, i| acc[i.name] = i; acc}
+
+    puts "===== IMPORTING USER TOKENS ====="
+
+    User.all.find_each(batch_size: BATCH_SIZE).each do |u|
+      user_kubernetes_identity = u.kubernetes_identity
+
+      data = user_kubernetes_identity.try(:data)
+      if data.blank? || !data.has_key?('tokens')
+        puts "-- Token data not present for user #{u.id}. Skipping user."
+        next
+      end
+
+      data['tokens'].each do |t|
+        cluster_name = t['cluster']
+
+        if clusters[cluster_name].nil?
+          puts "-- Token cluster with name `#{cluster_name}` doesn't exist (user #{u.id} has token defined for `#{cluster_name}` cluster). Skipping token."
+          next
+        end
+
+        begin
+          nt = user_kubernetes_identity.tokens.create!(
+            name: u.email,
+            cluster: clusters[cluster_name],
+            token: ENCRYPTOR.decrypt(t['token']),
+            uid: t['uid'],
+            groups: t['groups']
+          )
+
+          AuditService.log(
+            action: 'create',
+            auditable: nt,
+            data: {
+              cluster: cluster_name
+            }
+          )
+
+          puts "-- Token imported for user #{u.id} and cluster `#{cluster_name}`"
+        rescue => e
+          puts "-- Token import for user #{u.id} and cluster `#{cluster_name}` failed with #{e.message}"
+        end
+      end
+    end
+
+    puts "===== IMPORTING ROBOT TOKENS ====="
+
+    default_user = User.admin.first
```

Comment:
Should this be sorted by `created_at`?

Reply:
Assumed that any admin will be good enough, but could get the very first one if that makes the difference?

Reply:
Probably doesn't make a huge difference – fine as-is.